app/controllers/students_controller.rb

```diff
@@ -21,6 +22,7 @@ def edit
 
   def create
     @student = Student.new(student_params)
+    calculate_preferences(params)
```

Comment:
You'll want to do this before the `Student.new` line. That line will create a new `Student` but does not save it to the database yet so the params are already used and your calculations will not be used. If you're going to modify `params` then you'll also want to use `params` to create the object and not `student_params`.